### ISSUE TITLE
Update drupal/core to 8.4.x

### ISSUE THREAD

**Issue by balsama** [acquia/lightning#457]

**Comment by balsama** [comment]
Branch created: https://github.com/acquia/lightning/tree/8.4.x-2.x

Todo:
* Need to tweak some more tests to work with newer version of Drupal extension, specifically those that use scenario outlines.
* Updated to Drush 9 because of symfony/console and Drush 9 drops support for make-convert. However, we still need make convert to generate makefiles for D.O tarballs. We have officially deprecated the tarball since it doesn't include required composer dependnecies (specifically, league/oauth2-server), but we still need to package _something_ there in order to keep our [security advisory policy](https://www.drupal.org/drupal-security-team/security-advisory-process-and-permissions-policy) eligibility. So our plan is to extract the make convert command into a class that composer can call.
* Drush will likely still need to be patched to allow users to uninstall dependencies of parent profiles. I previously [submitted a patch for Drush 9](https://github.com/drush-ops/drush/pull/2698), but it no longer applies. So we'll need to reroll that.

**Comment by balsama** [comment]
As a reminder, all changes to Lightning (migration to core media, content moderation, adoption of the sidebar, etc) will be in subsequent releases. 2.2.0 will only update you to core 8.4.x.

**Comment by phenaproxima** [comment]
Here is a reroll of the patch from drush-ops/drush#2698.
[drush-ops--drush--9.x--2698.txt](https://github.com/acquia/lightning/files/1299765/drush-ops--drush--9.x--2698.txt)

**Comment by phenaproxima** [comment]
Regarding your first point: I think all will be well once jhedstrom/drupalextension tags version 3.3.1, which includes a fix (authored by me) that makes scenario outlines work correctly.

**Comment by balsama** [comment]
Great. I think we should pull in the patch for the issue against drupalextension then and pin. That way we can ensure it's going to work now and don't have to worry about whether they release or not. Hopefully we can remove it before we release.

**Comment by phenaproxima** [comment]
Your second request is now addressed in the 8.4.x-2.x branch. Our packaging script no longer relies on Drush at all, but it generates identical make files as before.

**Comment by phenaproxima** [comment]
@jhedstrom has tagged v3.3.1 of Drupal Extension, so I think we are probably in the clear to update. https://github.com/jhedstrom/drupalextension/releases/tag/v3.3.1

**Comment by balsama** [comment]
So we're down to four failing behat tests. Unit tests all pass 🍾 

Travis results here: https://travis-ci.org/acquia/lightning/jobs/278267888 Specifically, that is the 2.1.8 => HEAD test.

**/features/layout/panelizer.feature:29**
```
And I place the "block_content:test--here-be-dragons" block from the "Custom" category             # Acquia\LightningExtension\Context\PanelsInPlaceContext::placeBlock()
      Unable to complete AJAX request. (RuntimeException)
```
**/features/layout/panelizer_wizard.feature:83**
```
Then I should see a "block_content:test--here-be-dragons" block                           # PanelsSubContext::assertBlock()
      Element matching css "div[data-block-plugin-id = "block_content:test--here-be-dragons"]" not found. (Behat\Mink\Exception\ElementNotFoundException)
```

**/features/search.feature:20**
```
And I place the "block_content:dragons" block from the "Custom" category                                # Acquia\LightningExtension\Context\PanelsInPlaceContext::placeBlock()
      Unable to complete AJAX request. (RuntimeException)
```

**/features/workflow/moderation_states.feature:98**
```
Then I should see the "Save and publish" button                                                   # Drupal\DrupalExtension\Context\MinkContext::assertButton()
      The button 'Save and publish' was not found on the page http://127.0.0.1:8080/node/add/not_moderated (Exception)
```

**Comment by balsama** [comment]
Released as part of 2.2.0
